logicmonitor/resource_device.go

```diff
@@ -149,61 +157,78 @@ func updateDevice(d *schema.ResourceData, m interface{}) error {
-	restResponse, apiResponse, e := client.DeleteDevice(int32(deviceID))
-	err = checkStatus(restResponse.Status, restResponse.Errmsg, apiResponse.StatusCode, apiResponse.Status, e)
+	params := lm.NewDeleteDeviceByIDParams()
+	params.SetID(int32(deviceID))
+	restResponse, err := client.LM.DeleteDeviceByID(params)
 	if err != nil {
 		log.Printf("Error deleting device %s", err)
 		d.SetId("")
 	}
 
+	log.Printf("delete device response %v", restResponse.Payload)
+
 	d.SetId("")
 	return nil
 }
 
 // function to import devices
 func resourceDeviceStateImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	client := m.(*lmv1.DefaultApi)
+	client := m.(*lmclient.LMSdkGo)
 
 	// if it is a deviceId, we will add the group directly
 	if checkID(d.Id()) {
 		id, err := strconv.Atoi(d.Id())
 		if err != nil {
 			return nil, err
 		}
-		restResponse, apiResponse, e := client.GetDeviceById(int32(id), "")
-		err = checkStatus(restResponse.Status, restResponse.Errmsg, apiResponse.StatusCode, apiResponse.Status, e)
+		params := lm.NewGetDeviceByIDParams()
+		params.SetID(int32(id))
+
+		restResponse, err := client.LM.GetDeviceByID(params)
 		if err != nil {
 			log.Printf("Failed to find device %q", err)
 			return nil, err
 		}
-		d.Set("ip_addr", restResponse.Data.Name)
-		d.Set("collector", restResponse.Data.PreferredCollectorId)
-		d.Set("disable_alerting", restResponse.Data.DisableAlerting)
-		d.Set("description", restResponse.Data.Description)
-		d.Set("hostgroup_id", restResponse.Data.HostGroupIds)
+		d.Set("ip_addr", restResponse.Payload.Name)
+		d.Set("collector", restResponse.Payload.PreferredCollectorID)
+		d.Set("disable_alerting", restResponse.Payload.DisableAlerting)
+		d.Set("description", restResponse.Payload.Description)
+		d.Set("hostgroup_id", restResponse.Payload.HostGroupIds)
+		properties := make(map[*string]*string)
+		props := restResponse.Payload.CustomProperties
+		for _, v := range props {
+			properties[v.Name] = v.Value
+		}
+		d.Set("properties", properties)
 	} else {
 
 		// currently set to add displayname
-		filter := fmt.Sprintf("name:%s", d.Id())
-
-		//looks for device Group
-		restResponse, apiResponse, e := client.GetDeviceList("", -1, 0, filter)
-		err := checkStatus(restResponse.Status, restResponse.Errmsg, apiResponse.StatusCode, apiResponse.Status, e)
+		filter := fmt.Sprintf("displayName:\"%s\"", d.Id())
+		//looks for device
+		params := lm.NewGetDeviceListParams()
+		params.SetFilter(&filter)
+		restResponse, err := client.LM.GetDeviceList(params)
 		if err != nil {
 			return nil, err
 		}
 
-		if restResponse.Data.Total > 0 {
-			d.Set("ip_addr", restResponse.Data.Items[0].Name)
-			d.Set("display_name", restResponse.Data.Items[0].DisplayName)
-			d.Set("collector", restResponse.Data.Items[0].PreferredCollectorId)
-			d.Set("disable_alerting", restResponse.Data.Items[0].DisableAlerting)
-			d.SetId(strconv.Itoa(int(restResponse.Data.Items[0].Id)))
+		if restResponse.Payload.Total > 0 {
```

Comment:
same as my comments on collector group. don't just assume there's only one result. throw and error if Total > 1